src/charm.py

```diff
@@ -147,7 +156,7 @@ def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
     def _on_install(self, _: ops.InstallEvent) -> None:
         """Handle install."""
         self.unit.status = ops.MaintenanceStatus("Preparing bind")
```

Comment:
If something goes wrong in the setup...What will the status of the unit be? 
Would the collect status changing it for active even if something unexpected happens?

Reply:
Good question. I have to test that

Reply:
After testing I confirm that the status is updated for `hook install fail`, and the charm ends up in an error state